packages/addons-v5/commands/addons/wait.js

```diff
@@ -16,20 +17,32 @@ function * run (ctx, api) {
     } else {
       addons = yield api.get('/addons')
     }
-    addons = addons.filter(addon => addon.state === 'provisioning')
   }
+  addons = addons.filter(addon => addon.state === 'provisioning')
 
   let interval = parseInt(ctx.flags['wait-interval'])
   if (!interval || interval < 0) { interval = 5 }
 
   for (let addon of addons) {
-    addon = yield waitForAddonProvisioning(api, addon, interval)
+    const startTime = new Date()
+    try {
+      addon = yield waitForAddonProvisioning(api, addon, interval)
+    } catch (error) {
+      cli.error(error)
+      notify(`heroku addons:wait ${addon.name}`, 'Add-on failed to provision', false)
+      throw error
+    }
 
     let configVars = (addon.config_vars || [])
     if (configVars.length > 0) {
       configVars = configVars.map(c => cli.color.configVar(c)).join(', ')
       cli.log(`Created ${cli.color.addon(addon.name)} as ${configVars}`)
     }
+
+    // only show notification if addon took longer than 20 seconds to provision
```

Comment:
:thinking:

Why this exception? I guess you don't want to spam users for something that provisions quickly? I guess it makes sense, though 20s seems high as a threshold--if something is taking more than a couple of seconds, I'm likely to context-switch. I guess what we really want is "is the user still watching the provisioning command in the terminal" but no great way to check that.

Reply:
yeah I think I agree, maybe 5 seconds is more reasonable? We do something like this on `heroku run` waiting for the dyno to load, but when the user is running `addons:wait` I think a notifications is much less annoying

Reply:
@uhoh-itsmaciek I'm glad you brought this up, because I don't have much insight into what number makes the most sense here.  Is there any CLI UX guidelines around this, or should we go with 5 seconds?

Reply:
I think we need to establish these times on a case-by-case basis. It's going to be different for every action. I think for `*:wait` 5 seconds is a good place to start.